contracts/CBKLandSale.sol

```diff
@@ -217,16 +235,66 @@ contract CBKLandSale is Initializable, AccessControlUpgradeable {
         cbkLand.mint(buyer, TIER_THREE, chunkId);
     }
 
-    function giveT1LandReserved(address buyer) public  {
-        
+    function giveT1LandReserved(address buyer) public {
+        giveT1LandReserved(buyer, '');
     }
 
-    function giveT2LandReserved(address buyer, uint256 chunkId) public  {
-        
+    function giveT2LandReserved(address buyer, uint256 chunkId) public {
+        giveT2LandReserved(buyer, chunkId, '');
+    }
+
+    function giveT3LandReserved(address buyer, uint256 chunkId) public {
+        giveT3LandReserved(buyer, chunkId, '');
     }
 
-    function giveT3LandReserved(address buyer, uint256 chunkId) public  {
-      
+
+    function giveT1LandReserved(address buyer, string memory uri) public reservedSaleAllowed() {
+        uint256 rcLength = reservedChunks[msg.sender].length();
+        require(rcLength > 0, "no reserved chunks");
+        uint256 counter = reservedChunksCounter[msg.sender];
+        for(uint256 i = counter; i < counter + rcLength; i++){
+            uint256 cId = reservedChunks[msg.sender].at(uint256(i % rcLength));
+            if(chunkT2LandSales[cId] < _allowedLandSalePerChunk) {
+                chunkT2LandSales[cId]++; // it's actually a T1, but we will play on the T2 because population is shared
+                chunkZoneLandSales[chunkIdToZoneId(cId)]++;
+                cbkLand.mint(buyer, TIER_ONE, cId, uri);
+                emit T1GivenReserved(msg.sender, buyer, cId);
+                reservedChunksCounter[msg.sender] = uint256(i + 1);
+                return;
+            }
+        }
+
+        // Could not find a land
+        revert();
+    }
+
+    function giveT2LandReserved(address buyer, uint256 chunkId, string memory uri) public reservedChunkAvailable(chunkId) reservedSaleAllowed() {
+        if(chunkT2LandSales[chunkId] == 0){
+            chunksWithT2Land++;
+        }
+
+        chunkT2LandSales[chunkId]++;
+        chunkZoneLandSales[chunkIdToZoneId(chunkId)]++;
+
+        purchaseAddressMapping[buyer] = purchaseInfo(buyer, TIER_TWO, chunkId, false);
```

Comment:
is this operation missing from the two other reserve functions? Or is this mistakenly present here and neither 3 are meant to have it

Reply:
correct, not sure why I keep missing these. Removing.